add-database

ilya-noize opened the pull request:
Создание базы данных
Работа с DAO
Тестирование
Доработка бизнес-логики
Финальный кадр

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

friends in memory;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

refactoring storages and services;
add something exceptions;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2
add friends in users;
add likes in films;

ilya-noize committed:
Pull Request #2

fix checkstyle;

ilya-noize committed:
Pull Request #2

fix checkstyle;

ilya-noize committed:
Pull Request #2

fix tests;

ilya-noize committed:
Pull Request #2

fix constructors;
renaming;

ilya-noize committed:
Pull Request #2

POSTMAN Film tests Ok.

ilya-noize committed:
Pull Request #2

Delete Controller.java
patch for test postman "User id=2 get 2 friends" (not worked)

ilya-noize committed:
Pull Request #2

fix checkstyle;

ilya-noize committed:
Pull Request #2

fix checkstyle;

ilya-noize committed:
Pull Request #2

friends in User.class

ilya-noize committed:
Pull Request #2

after review

ilya-noize committed:
Pull Request #2

after review. remaster after night.

ilya-noize committed:
Merge remote-tracking branch 'origin/add-friends-likes' into add-friends-likes

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/ErrorController.java
#	src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java
#	src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java
#	src/main/java/ru/yandex/practicum/filmorate/exception/EmptyFriendsSetException.java
#	src/main/java/ru/yandex/practicum/filmorate/model/Film.java
#	src/main/java/ru/yandex/practicum/filmorate/model/User.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmServiceImpl.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserServiceImpl.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/InMemoryFilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/user/InMemoryUserStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/user/UserStorage.java
#	src/test/java/ru/yandex/practicum/filmorate/controller/FilmControllerTest.java
#	src/test/java/ru/yandex/practicum/filmorate/controller/UserControllerTest.java

ilya-noize committed:
Pull Request #2

after review. remaster after night-day.

ilya-noize committed:
Pull Request #2

after review. abstract class MainStorage<T>
try mockMvc

ilya-noize committed:
Pull Request #2
abstract class MainStorage<T> for Films, Users;
abstract class MainService<T> for Films, Users;

APPLICATION FAILED TO START
Description:
Parameter 1 of constructor in ru.yandex.practicum.filmorate.service.FilmLikesService required a bean of type 'ru.yandex.practicum.filmorate.service.MainService' that could not be found.

Action:
Consider defining a bean of type 'ru.yandex.practicum.filmorate.service.MainService' in your configuration.

ilya-noize committed:
Pull Request #2
abstract class MainStorage<T> for Films, Users;
abstract class MainService<T> for Films, Users;

APPLICATION FAILED TO START
Description:
Parameter 1 of constructor in ru.yandex.practicum.filmorate.service.FilmLikesService required a bean of type 'ru.yandex.practicum.filmorate.service.MainService' that could not be found.

Action:
Consider defining a bean of type 'ru.yandex.practicum.filmorate.service.MainService' in your configuration.

fix checkstyle;

ilya-noize committed:
Pull Request #2
abstract class MainStorage<T> for Films, Users;
abstract class MainService<T> for Films, Users;

APPLICATION FAILED TO START
Description:
Parameter 1 of constructor in ru.yandex.practicum.filmorate.service.FilmLikesService required a bean of type 'ru.yandex.practicum.filmorate.service.MainService' that could not be found.

Action:
Consider defining a bean of type 'ru.yandex.practicum.filmorate.service.MainService' in your configuration.

fix checkstyle;

ilya-noize committed:
Pull Request #2
abstract class MainStorage<T> for Films, Users;
abstract class MainService<T> for Films, Users;

APPLICATION FAILED TO START
Description:
Parameter 1 of constructor in ru.yandex.practicum.filmorate.service.FilmLikesService required a bean of type 'ru.yandex.practicum.filmorate.service.MainService' that could not be found.

Action:
Consider defining a bean of type 'ru.yandex.practicum.filmorate.service.MainService' in your configuration.

fix checkstyle;

ilya-noize committed:
Pull Request #2
abstract class MainStorage<T> for Films, Users;
abstract class MainService<T> for Films, Users;

APPLICATION FAILED TO START
Description:
Parameter 1 of constructor in ru.yandex.practicum.filmorate.service.FilmLikesService required a bean of type 'ru.yandex.practicum.filmorate.service.MainService' that could not be found.

Action:
Consider defining a bean of type 'ru.yandex.practicum.filmorate.service.MainService' in your configuration.

fix checkstyle;

ilya-noize committed:
Pull Request #2
abstract class MainStorage<T> for Films, Users;
abstract class MainService<T> for Films, Users;

APPLICATION FAILED TO START
Description:
Parameter 1 of constructor in ru.yandex.practicum.filmorate.service.FilmLikesService required a bean of type 'ru.yandex.practicum.filmorate.service.MainService' that could not be found.

Action:
Consider defining a bean of type 'ru.yandex.practicum.filmorate.service.MainService' in your configuration.

fix checkstyle;

ilya-noize committed:
Pull Request #2
abstract class MainStorage<T> for Films, Users;
abstract class MainService<T> deleted;
edits after the review:
 - 'required="10"' deleted;
 - encapsulating a variable;
 - rename MainStorage to MasterStorage
 - private final Map<Long, T> storage in UserStorage and FilmStorage

ilya-noize committed:
Pull Request #2
abstract class MasterService<T> for Films, Users remastered;

ilya-noize committed:
Pull Request #2
logging requests in create, update-methods
translate in english

ilya-noize committed:
Pull Request #2
In each handler added logging stack trace errors
translate in english

ilya-noize committed:
Pull Request #2
catch Throwable with HttpStatus.500

ilya-noize committed:
Pull Request #2
Update README.md

ilya-noize committed:
Pull Request #2
edits after review. Merge branch with main

ilya-noize committed:
Modify pom.xml:
 add dependency 'spring-boot-starter-data-jpa' from 'spring-boot'
 add 'h2' from 'com.h2database'
Modify application.properties (connect to DB H2 file:filmorate.mv.db);
Create database connection;
Add schema.sql, data.sql;

Add Models: Genre, MPARating;
Add Data Access Objects: Film, User, Genre, MPARating;

ilya-noize committed:
remove dir with DAO: Film, User, Genre, MPARating;

ilya-noize committed:
add logging in create-, update-methods

ilya-noize committed:
fix checkstyles

ilya-noize committed:
Schema.drawio

ilya-noize committed:
tests switch off

ilya-noize committed:
update schema database;

ilya-noize committed:
Simply 'friends'.
Without status friendship;

ilya-noize committed:
+ Apache Freemarker

ilya-noize committed:
prepare database to working

ilya-noize committed:
create Friends (Model, DAO, Service);

ilya-noize committed:
add/refactor Genre (Model, DAO, Service);

ilya-noize committed:
RowMappers: MPARating, GenresFilm;

ilya-noize committed:
add GenresFilm (Model, DAO, Service);

ilya-noize committed:
add Like (Model, DAO, Service);

ilya-noize committed:
MasterStorage + MasterService {
 add delete(Long... id);
 add isExist(Long id);
}
MasterStorageDAO ext MasterStorage {
 support DAO {
  private JdbcTemplate;
  add Getter:getjdbcTemplate;
 }
 add make(rs, rowNum);
}

ilya-noize committed:
Adapted by .ElseThrow(*)

ilya-noize committed:
User passed Postman tests;

ilya-noize committed:
add/refactor MPARating (Model, DAO, Service, Controller);

ilya-noize committed:
UserControllerTest switch off;

ilya-noize committed:
final class

ilya-noize committed:
short logging;
add/del likes;
add/del genres;

ilya-noize committed:
refactoring.
add features:
 update
 get
 delete
 make
[failed Postman test!]

ilya-noize committed:
refactoring.
getPopular [failed Postman test!]

ilya-noize committed:
refactoring.
constructor [failed Postman test!];
rename fields;
final class;

ilya-noize committed:
add delete-method;
fix create-method;

ilya-noize committed:
schema [PDF, PNG, (mysql)SQL]

ilya-noize committed:
fixed for checkstyle test

ilya-noize committed:
fixed for checkstyle test

ilya-noize committed:
private Set<Genre> genres;
With all the consequences of this:
 add GenreRowMapper;
 refactoring FilmDAO

ilya-noize committed:
FIlm: rate = null; // switch off
fixed genres;
clean code;

ilya-noize committed:
refactor addLike

ilya-noize committed:
FilmRowMapper include in FilmDAO;
fix field genres;
clean code;

ilya-noize committed:
GenreRowMapper include in GenreDAO;
`GROUP BY ID` in `getAll`

ilya-noize committed:
GenresFilmRowMapper include in GenresFilmDAO
`GROUP BY ID` in `getAll`

ilya-noize committed:
The composite key in the `films_like` table is disabled

ilya-noize committed:
The composite key in the `films_like` table is enabled;

ilya-noize committed:
+ `delete from genres_film;`

ilya-noize committed:
final class

ilya-noize committed:
refactoring

ilya-noize committed:
fixeg update Genres of Film;

ilya-noize committed:
refactoring

ilya-noize committed:
refactoring:
update switch off;

ilya-noize committed:
switch on MasterStorageDAO;

ilya-noize committed:
switch on MasterStorageDAO;

ilya-noize committed:
switch on MasterStorageDAO;

ilya-noize committed:
switch on MasterStorageDAO;

ilya-noize committed:
refactoring

ilya-noize committed:
fix

ilya-noize committed:
toSet in import static

ilya-noize committed:
delete create increment

ilya-noize committed:
refactoring

ilya-noize committed:
reversed Popular Films

ilya-noize committed:
comeback increment %)

ilya-noize committed:
delete genre-mapping

ilya-noize committed:
refactoring
move in service

ilya-noize committed:
override methods:
 update
 get
 getAll

ilya-noize committed:
searching bag with Genres Film

ilya-noize committed:
rename field

ilya-noize committed:
rename field

ilya-noize committed:
rename field

ilya-noize committed:
logging
move code working with DB from service to DAO
update genres for films ( GenresFilmDAO )
refactoring

ilya-noize committed:
refactoring LikesFilm

ilya-noize committed:
unq_friends UNIQUE (user_id_request, user_id_friend);
unq_likes UNIQUE (film_id, user_id);

ilya-noize committed:
Set -> List

ilya-noize committed:
Autoincrement + Validation

ilya-noize committed:
spring-boot-starter-test incoming junit

ilya-noize committed:
rollover commit

ilya-noize committed:
rollover commit

ilya-noize committed:
refactoring ErrorController.
fix bugs in User/Friends/FilmService.

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    private List<Genre> getGenresByFilm() {
+        String sql = "SELECT G.ID, G.NAME FROM GENRES_FILM GF"
```
На уровне бизнес-логики не должно быть работы с SQL, так как это должно происходят на уровне доступа к данным)

ilya-noize replied:
Согласен. Только чёт не понял как достать жанры фильмов из таблицы где лежать их id. 
Предыдущее решение было ещё более тяжеловесное на двух стримах. )

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    private List<Long> getUserLikes() {
+        String sql = "SELECT user_id FROM films_like WHERE film_id = ? ORDER BY user_id";
```
На уровне бизнес-логики не должно быть работы с SQL, так как это должно происходят на уровне доступа к данным)

TyutterinYakov reviewed (commented):
Добрый день! В целом - хорошо) Отметил то, что следует исправить) Хорошего дня! Если возникнут вопросы - пиши в пачку, буду рад помочь🌅

ilya-noize commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -1,54 +1,40 @@
+    @Past(message = "The date of birth should only be in the past.")
     private LocalDate birthday;
+    private Set<Long> friends;
+    private Set<Long> likes;
```
)

ilya-noize commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,156 @@
+        return film;
+    }
+
+    @NotNull(message = "The name of the movie cannot be null.")
```
Учитываются, вроде. Проверю ещё раз.

ilya-noize commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,156 @@
+        return film.getName();
+    }
+
+    @CorrectReleaseDate(value = RELEASE_DATE_LIMIT, message = "The release date is not earlier than a certain date and not later than today")
```
Учитываются, вроде. Проверю ещё раз.

ilya-noize commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,156 @@
+        return film.getReleaseDate();
+    }
+
+    @Size(max = 200, message = "The description length is no more than 200 characters.")
```
Учитываются, вроде. Проверю ещё раз.

ilya-noize committed:
refactoring model Film

ilya-noize committed:
refactoring model User

ilya-noize committed:
refactoring model {Genre, MPARating};

ilya-noize committed:
refactoring model {Friends, LikesFilm};

ilya-noize committed:
refactor: delete comments, add 9 genres, UpperCase;

ilya-noize committed:
Warning: Immutable object is modified;

ilya-noize committed:
destroy old files from Sprints 9-10.
in Spring 11 realize on DAO.

ilya-noize committed:
destroy all notNeeded/renamed files.

ilya-noize committed:
refactor:
 removed the abstraction structure;
 made query optimization;

ilya-noize committed:
refactor:
 changed the class name and field

ilya-noize committed:
refactor:
 removed the abstraction structure;
 made query optimization;

ilya-noize committed:
refactor:
 added serialization of the modified class field

ilya-noize committed:
refactor:
 a new implementation of supporting genres of films

ilya-noize committed:
clean code;

ilya-noize committed:
refactor:
 delete serialization of the modified class field
 changed the constructor of the class during serialization

ilya-noize committed:
refactor:
 removed the abstraction structure;
 moving some methods to the DAO layer;
 clean code;

ilya-noize committed:
refactor:
 clean code;

ilya-noize committed:
refactor:
 removed the abstraction structure;
 made query optimization;
 methods have been created for working with the `friends` table

ilya-noize committed:
feature:
 create RowMapper for model: Users;

ilya-noize committed:
refactor:
 removed the abstraction structure;
 methods have been created for working with the `friends` table;
 clean code;

ilya-noize committed:
refactor:
 clean code;

ilya-noize committed:
refactor:
 rename tables

ilya-noize committed:
refactor:
 removed the abstraction structure;
 methods have been created for working with the `mpa_rating` table;
 clean code;

ilya-noize committed:
refactor:
 @Component

ilya-noize committed:
refactor:
 removed the abstraction structure;
 methods have been created for working with the `mpa_rating` table;
 clean code;

ilya-noize committed:
refactor:
 changed logging;

ilya-noize committed:
refactor:
 removed the abstraction structure;
 methods have been created for working with the `FILMS_LIKE` table;
 clean code;

ilya-noize committed:
refactor:
 removed the abstraction structure;
 methods have been created for working with the `GENRES` table;

ilya-noize committed:
refactor:
 @Component

ilya-noize committed:
refactor:
 removed the abstraction structure;
 methods have been created for working with the `GENRES` table;
 clean code;

ilya-noize committed:
fix: changed the processing of likes;

ilya-noize committed:
destroy all notNeeded/renamed files.

ilya-noize committed:
refactor: the last 9 genres as a comment for passing tests (._.);

ilya-noize committed:
fix: incorrect response by endpoint /genres/{id}

ilya-noize committed:
docs: edit README.md

ilya-noize committed:
test: create sql files

ilya-noize committed:
test: service MPA Rating - ok

ilya-noize committed:
exception:
 add ValidException;
 remove FailSetFilmLikesException;
 remove FailSetFriendException;

ilya-noize committed:
fix: add film validation;

ilya-noize committed:
code: fix constructor;

ilya-noize committed:
refactor: add getFilmGenres;

ilya-noize committed:
fix: UTF-8 in tests

ilya-noize committed:
refactor: delete 'agree' in friends;

ilya-noize committed:
test: controller Genre - ok

ilya-noize committed:
fix sql

ilya-noize committed:
test: service Genre - ok

ilya-noize committed:
refactor: sql-files on folders

ilya-noize committed:
fix: path sql

ilya-noize committed:
feature: generate sql insert into from arrays;

ilya-noize committed:
test: controller Film - ok

ilya-noize committed:
test: service MPA Rating - ok

ilya-noize committed:
fix sql

ilya-noize committed:
test.properties

ilya-noize committed:
fix: change AllArgsConstructor

ilya-noize committed:
fix: change AllArgsConstructor

ilya-noize committed:
refactor: remove FilmGenresDAO

ilya-noize committed:
refactor: remove getFilmGenres

ilya-noize committed:
refactor: html

ilya-noize committed:
test: new test

ilya-noize committed:
refactor: remove FilmGenresDAO from constructor

ilya-noize committed:
fix: change AllArgsConstructor

ilya-noize committed:
refactor: clean code;

ilya-noize committed:
fix: change AllArgsConstructor by Film

ilya-noize committed:
fix: change AllArgsConstructor

ilya-noize committed:
fix: change filmGenres methods

ilya-noize committed:
refactor: remove GenreDAO from Service-constructor

ilya-noize committed:
refactor: clean code;

ilya-noize committed:
refactor: log.debug

ilya-noize committed:
refactor: final fields but not mpa,genres

ilya-noize committed:
refactor: final fields

ilya-noize committed:
refactor: final fields

ilya-noize committed:
refactor: final fields

ilya-noize committed:
refactor: new Interfaces

ilya-noize committed:
refactor: implements

ilya-noize committed:
refactor: log.debug;
 fields class by interfaces;

ilya-noize committed:
refactor: implements Showable

ilya-noize committed:
refactor: log.debug

ilya-noize committed:
refactor: final fields;
delete custom constructor;

ilya-noize committed:
refactor: clean code;

ilya-noize committed:
test: go UserControllerTest

ilya-noize committed:
refactor: log.debug

ilya-noize committed:
refactor: implements Showable

ilya-noize committed:
refactor: clean code;

ilya-noize committed:
refactor: clean code;

ilya-noize committed:
test: sql mpaRating

ilya-noize committed:
update ignore paths

ilya-noize committed:
fix test

ilya-noize committed:
fix test

ilya-noize committed:
update schema db

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -0,0 +1,157 @@
+        String login = user.getLogin();
+
+        final Predicate<User> TWIN_LOGIN = user1 -> user1.getLogin().equals(login);
+        Optional<User> result = getAll().stream().filter(TWIN_LOGIN).findFirst();
```
Эта логика не совсем эффективна, так как таким образом мы производим обращение к БД, извлекая все записи) Хорошо было бы описать условие поиска в запросе к БД

Но и это не требуется, так как на уровне таблицы БД установлен индекс для поддержания уникальности логина, а следовательно будет выброшено исключение, если запись будет присутствовать)
Это исключение можно будет обработать в одном из хендлеров и вернуть требуемый статус)

Текущая же логика не совсем корректна, так как при обновлении может быть передан логин соответствующий тому, что был до обновления, а значит будет выброшено исключение, чего не должно было произойти)

TyutterinYakov commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,149 @@
+            } else {
+                film.setGenres(genres.get(film.getId()));
+            }
+            film.setMpa(mpaRatingDAO.get(film.getMpa().getId()));
```
При текущей реализации мы плодим n-лишние запросы, так как производим обращение к методу репозитория в цикле) Эту логику можно исключить, так как теперь таблицу с рейтингами мы будем присоединять в запросах при получении фильма/фильмов, а затем извлекать их в `resultSet`

TyutterinYakov reviewed (commented):
Добрый день! Отметил оставшиеся недочеты) Хорошего дня! Если возникнут вопросы - пиши в пачку, буду рад помочь🌅

ilya-noize committed:
deleting files that have lost their meaning

ilya-noize committed:
add logging

ilya-noize committed:
code: delete @primary

ilya-noize committed:
refactor: optimize requests;
 valid body in comments;

ilya-noize committed:
refactor: by interfaces;

ilya-noize committed:
refactor: new interface<generic>;

ilya-noize committed:
refactor: rename table columns;
 extract mpa_rating;

ilya-noize committed:
refactor: add HttpStatus in map;

ilya-noize committed:
refactor: optimize requests;

ilya-noize committed:
fix: getPopular bag

ilya-noize committed:
fix test

ilya-noize committed:
fix: remake isLoginUnique;
 to bo continue...

ilya-noize committed:
code: delete valid-method.

ilya-noize committed:
code: delete valid-method.

ilya-noize committed:
Merge remote-tracking branch 'origin/add-database' into add-database

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/test/java/ru/yandex/practicum/filmorate/service/FilmServiceTest.java

ilya-noize committed:
refactor: Polymorphism in Services;
 remake isExist in DAO: User, Film;
 remake UniqueLogin in DAO: User;
 add annotation in Model: User, Film;
 docs in tests;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
code: delete valid-method.

ilya-noize committed:
fix checkstyle;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize commented on src/main/java/ru/yandex/practicum/filmorate/controller/ErrorController.java:
```diff
@@ -0,0 +1,84 @@
+
+    @ExceptionHandler({BindException.class, MethodArgumentNotValidException.class})
+    public ResponseEntity<Map<String, Object>> handleException(BindException e) {
+
```
+

ilya-noize committed:
refactor: NotFoundException in MPARatingDAO and GenreDAO in lambda;
 delete Supplier<> interface in class;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
fix: add check isNull on isValid

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
fix: add @NotNull on fields

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
fix: Validation count

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
refactor: FilmLikesDAO Interface;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
refactor: batchUpdate;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
clean code;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
fix COLUMN ID RESTART with .. before testing;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
refactor: Exceptions return ResponseEntity;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
refactor: change Exception;
 renaming methods;
 add successful-create tests;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize committed:
fix: logging;
 new Data Integrity Violation Exception

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

ilya-noize reviewed (commented):
All of your conversations have been resolved.

ilya-noize committed:
fix: expected status;

Signed-off-by: ilya-noize <shumakoff.ilya@yandex.ru>

TyutterinYakov reviewed (approved):
Добрый день! Поздравляю с успешной сдачей спринта :) Сил и терпения на этом сложном, но интересном пути! Успехов в обучении и продуктивной работы над групповым проектом :) Прекрасного дня и отличного настроения!🌅🎇

ilya-noize commented:
Разобрался и применил на практике работу с БД , в том числе класс JdbcTemplate.
Научился настраивать подключение к БД для спринга.
Освоил принципы работы с БД, которые отличаются от массивов эффективным использованием ресурсов базы данных.
Разобрался и применил на практике обработку исключений.
